frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -625,105 +681,136 @@ pub mod pallet {
             ledger.unbonding_info = future_chunks;
 
             Self::update_ledger(&staker, ledger);
+            EraRewardsAndStakes::<T>::mutate(&current_era, |value| {
+                if let Some(x) = value {
+                    x.locked = x.locked.saturating_add(withdraw_amount)
+                }
+            });
 
             Self::deposit_event(Event::<T>::Withdrawn(staker, withdraw_amount));
 
             Ok(().into())
         }
 
-        /// Claim the rewards earned by contract_id.
-        /// All stakers and developer for this contract will be paid out with single call.
-        /// claim is valid for all unclaimed eras but not longer than history_depth().
-        /// Any reward older than history_depth() will go to Treasury.
-        /// Any user can call this function.
-        #[pallet::weight(T::WeightInfo::claim(T::MaxNumberOfStakersPerContract::get() + 1))]
-        pub fn claim(
+        // TODO: do we need to add force methods or at least methods that allow others to claim for someone else?
+
+        /// Claim earned staker rewards for the oldest era.
+        #[pallet::weight(T::WeightInfo::claim_staker())]
+        pub fn claim_staker(
+            origin: OriginFor<T>,
+            contract_id: T::SmartContract,
+        ) -> DispatchResultWithPostInfo {
+            ensure!(!Self::pallet_disabled(), Error::<T>::Disabled);
+            let staker = ensure_signed(origin)?;
+
+            // Ensure we have something to claim
+            let mut staker_info = Self::staker_info(&staker, &contract_id);
+            let (era, staked) = staker_info.claim();
+            ensure!(staked > Zero::zero(), Error::<T>::NotStakedContract);
```

Comment:
`staked != Zero::zero()` maybe?

Reply:
I'd prefer `!staked.is_zero()` but the when we got audited by Quantstamp, they explicitly told me to prefer **is greater than** check.